pydicom/sequence.py

```diff
@@ -93,32 +107,57 @@ def insert(    # type: ignore[override]
     ) -> None:
         """Insert a :class:`~pydicom.dataset.Dataset` into the sequence."""
         super().insert(position, val)
-        val.parent = self._parent
+        val.parent_seq = self  # type: ignore
 
     @property
-    def parent(self) -> "Optional[weakref.ReferenceType[Dataset]]":
+    def parent_dataset(self) -> "Optional[weakref.ReferenceType[Dataset]]":
         """Return a weak reference to the parent
         :class:`~pydicom.dataset.Dataset`.
 
-        .. versionadded:: 1.3
-
-        .. versionchanged:: 1.4
+        .. versionadded:: 2.4
 
             Returned value is a weak reference to the parent ``Dataset``.
         """
-        return self._parent
+        return self._parent_dataset
+
+    @parent_dataset.setter
+    def parent_dataset(self, value: Dataset) -> None:
+        """Set the parent :class:`~pydicom.dataset.Dataset`
+
+        .. versionadded:: 2.4
+        """
+        if value != self._parent_dataset:
+            self._parent_dataset = weakref.ref(value)  # type: ignore
+
+    @property
+    def parent(self) -> "Optional[weakref.ReferenceType[Dataset]]":
+        """Return a weak reference to the parent Dataset
+
+        .. deprecated:: 2.4
+        """
+        if config._use_future:
+            raise AttributeError("Future: Sequence.parent is removed in v3.x")
+        else:
+            warnings.warn(
+                "Sequence.parent will be removed in pydicom 3.0",
+                DeprecationWarning
+            )
+            return self.parent_dataset
 
     @parent.setter
-    def parent(self, value: Dataset) -> None:
-        """Set the parent :class:`~pydicom.dataset.Dataset` and pass it to all
-        :class:`Sequence` items.
+    def parent(self, value: "Dataset") -> None:
+        """Set the parent :class:`~pydicom.dataset.Dataset`
 
-        .. versionadded:: 1.3
+        .. deprecated:: 2.4
         """
-        if value != self._parent:
-            self._parent = weakref.ref(value)
-            for item in self._list:
```

Comment:
So I guess this was a performance problem...

Reply:
Yes, when sequences accessed through an index, this was called on every index.  Even if accessed only once, it is a 'costly' operation time-wise for heavily nested large lists.